fetch_puzzle_stats.py

```diff
@@ -1,137 +1,170 @@
-        PUZZLE_INFO.format(date=date),
-        cookies={
-            'NYT-S': cookie,
-        },
+        f"{PUZZLE_DETAIL}/{puzzle_id}.json", cookies={"NYT-S": cookie}
     )
+
     puzzle_resp.raise_for_status()
-    puzzle_date = datetime.strptime(date, DATE_FORMAT)
-    puzzle_info = puzzle_resp.json().get('results')[0]
-    solve_resp = requests.get(
-        SOLVE_INFO.format(game_id=puzzle_info['puzzle_id']),
-        cookies={
-            'NYT-S': cookie,
-        },
-    )
-    solve_resp.raise_for_status()
-    solve_info = solve_resp.json().get('results')
-
-    solved = solve_info.get('solved', False)
-    checked = 'firstChecked' in solve_info
-    revealed = 'firstRevealed' in solve_info
-    solve_date = datetime.fromtimestamp(solve_info.get('firstSolved', 0))
-    # A puzzle is streak eligible if they didn't cheat and they solved it
-    # before midnight PST (assume 8 hour offset for now, no DST)
-    streak_eligible = solved and not checked and not revealed and (
-        solve_date <= puzzle_date + timedelta(days=1) + timedelta(hours=8))
-
-    return {
-        'elapsed_seconds': solve_info.get('timeElapsed', 0),
-        'solved': int(solved),
-        'checked': int(checked),
-        'revealed': int(revealed),
-        'streak_eligible': int(streak_eligible),
-    }
+    puzzle_detail = puzzle_resp.json()["calcs"]
 
+    # crude rate limiting
+    time.sleep(0.25)
+    return puzzle_detail
 
-if __name__ == '__main__':
+
+if __name__ == "__main__":
     args = parser.parse_args()
-    cookie = os.getenv('NYT_COOKIE')
+    cookie = os.getenv("NYT_COOKIE")
     if not cookie:
         cookie = login(args.username, args.password)
+
     start_date = datetime.strptime(args.start_date, DATE_FORMAT)
     end_date = datetime.strptime(args.end_date, DATE_FORMAT)
-    print("Getting stats from {} until {}".format(
-        datetime.strftime(start_date, DATE_FORMAT),
-        datetime.strftime(end_date, DATE_FORMAT)))
+
+    days_between = (end_date - start_date).days
+    batches = (days_between // 100) + 1
+
+    print(
+        f"Getting stats from {args.start_date} until {args.end_date} in {batches} batches"
+    )
+
     date = start_date
+
+    if end_date - start_date > timedelta(days=100):
+        batch_end = start_date + timedelta(days=100)
+    else:
+        batch_end = end_date
+    batch_start = start_date
+
+    puzzle_overview = []
+
+    for batch in (pbar := tqdm(range(batches))):
+        pbar.set_description(f"Start date: {batch_start}")
+        batch_overview = get_v3_puzzle_overview(
+            puzzle_type=args.type,
+            start_date=batch_start,
+            end_date=batch_end,
+            cookie=cookie,
+        )
+        puzzle_overview.extend(batch_overview)
+        batch_start = batch_start + timedelta(days=100)
+        batch_end = batch_end + timedelta(days=100)
+
+    # A temporary file just in case it dies with those solve times
+    with open("tmp.pkl", "wb") as f:
```

Comment:
Do you ever read from this? I like the idea of a cache in case of failure, but doesn't look like it's actually used so probably can be removed

Reply:
the code doesn't read it - but I had a bunch of failures in that second loop that gets solve times and got really frustrated. We can pull it out unless you want to somehow offer it as a "better-than-nothing" helper